org.spoofax.jsglr/src/org/spoofax/jsglr/client/imploder/IToken.java

```diff
@@ -93,5 +78,21 @@
 
     ITokens getTokenizer();
 
+    /**
+     * @return A token immediately preceding the current token, or null if the current token is the start token. <br>
+     *         If the preceding AST node is ambiguous, any of the possible tokens will be returned.
+     */
+    IToken getTokenBefore();
+
+    /**
+     * @return A token immediately following the current token, or null if the current token is the end token. <br>
+     *         If the following AST node is ambiguous, any of the possible tokens will be returned.
+     */
+    IToken getTokenAfter();
+
+    Collection<IToken> getTokensBefore();
+
+    Collection<IToken> getTokensAfter();
```

Comment:
Right, I forgot your comment below:

> These methods are not properly implemented in JSGLR1 yet, but I've added TODO comments there. I didn't want to do this because there is no proper way to test them, and they'll probably not be used soon

Why are they necessary?

Reply:
You're right, they're not used yet, but I was under the impression that they might come in handy some day. If you think that's not the case, I can also remove them.